app/components/Form/ApplicationForm.tsx

```diff
@@ -442,13 +453,19 @@ const ApplicationForm: React.FC<Props> = ({
       } else if (!isSubmitEnabled && !projectAreaAccepted) {
         setProjectAreaModalType('pre-submitted');
       }
-
-      setProjectAreaOpen(!projectAreaAccepted);
-
       setProjectAreaModalOpen(
         !projectAreaAccepted &&
           (geographicAreaInputChanged || firstNationsLedInputChanged)
       );
+      projectAreaAccepted =
+        firstNationsLed ||
```

Comment:
Why are we re-assigning this here? I'm a bit confused as to the purpose here. If this is due to the case of it already being submitted, I'd put this in the `if` statement above. I also don't think we need to separate `if (isSubmitted && (isGeographicAreaEmpty || !isProjectAreaAccepted))` and `if (isSubmitted && isProjectAreaAccepted)` since isProjectAreaAccepted should be false if `geographicArea` is either undefined or empty.

Additionally, I know this isn't part of the ticket and will be put for a new ticket, but the logic here is getting a bit much and could use a refactor/some comments.

Reply:
Yes I am reassigning to avoid erroneous validation errors showing(eg: submit page)
but yes you are right, we can combine `isGeographicAreaEmpty || !isProjectAreaAccepted together.

I am not sure if we can add the recalculation logic inside if statement, cause we still need to handle modal logic after all the changes, but anyways extracted to another variable since we are changing isProjectAreaAccepted logic with new changes and for clarity as well.

Reply:
I meant moreso because we're doing this calculation for the portion surrounding the submitted portion, that re-check should be in that if statement surrounding the submitted check and encapsulating that logic in one area.

Reply:
yes that is correct, but if we do so, won't we loose the correct functionality that we need for `setProjectAreaModalOpen`?
but anyways, with new changes I have kind of separated the `isProjectAreaAccepted` from `isProjectAreaSelected`